pkg/api/types.go

```diff
@@ -153,6 +153,15 @@ const (
 	TerminationMessagePathDefault string = "/dev/termination-log"
 )
 
+// There is no namespace required for these kinds specified here.
+// If a kind is not enumerated here, it is assumed to have a namespace scope.
+var NoneNamespacedKinds = []string{
```

Comment:
Yeah, that doesn't bode well for this PR.  To do this sanely (at least as I can see it) involves doing something like making a validator context which the validation functions take (as an arg or as a method receiver).  That's no longer a small fix.

@lavalamp since he had some thoughts about validation perhaps being 2 passes - one to validate the object with no external state and another to validate against other objects or the system state.  This sort of falls in the middle..